dlk/python/dlk/templates/include/func/extract_image_patches.h

```diff
@@ -23,6 +23,61 @@ limitations under the License.
 #include "pack_input_to_qwords.h"
 #include <limits.h>
 
+inline void func_ExtractImagePatches(
+    const TensorView<QUANTIZED_PACKED, MemoryLayout::HWChBCl>& input,
+    const TensorView<QUANTIZED_PACKED, MemoryLayout::HWChBCl>& output,
+    T_UINT kernel_size, T_UINT stride) {
+  Measurement::Start("ExtractImagePatches");
+
+  const auto in_shape = input.get_shape();
+  const T_UINT input_width = in_shape[1];
+  const T_UINT input_depth = in_shape[2];
+  const T_UINT bits_per_input = in_shape[3];
+  const auto out_shape = output.get_shape();
+  const T_UINT out_height = out_shape[0];
+  const T_UINT out_width = out_shape[1];
+  const T_UINT out_depth = out_shape[2];
+
+  T_UINT output_index = 0;
+
+  if (out_depth < kernel_size * kernel_size) {
+    int bit_shift = out_depth * QUANTIZED_PACKED::BitCount / (kernel_size * kernel_size);
+    const QUANTIZED_PACKED::base_t mask((QUANTIZED_PACKED::base_t(1) << bit_shift) - 1);
+    std::fill(output.data(), output.data() + output.size(), QUANTIZED_PACKED(0));
+    for(T_UINT wi = 0; wi < out_height; wi++)
+      for(T_UINT wj = 0; wj < out_width; wj++)
+        for(T_UINT ki = 0; ki < kernel_size; ki++)
+          for(T_UINT kj = 0; kj < kernel_size; kj++)
+          {
+            T_INT row = (wi * stride) + ki;
+            T_INT col = (wj * stride) + kj;
+            T_UINT ch = (ki * kernel_size + kj) * bit_shift;
+            T_UINT ch_high = ch / QUANTIZED_PACKED::BitCount;
+            T_UINT ch_low = ch % QUANTIZED_PACKED::BitCount;
+            for(T_UINT digit = 0; digit < bits_per_input; ++digit) {
+              output(wi, wj, ch_high, digit, 0) |= QUANTIZED_PACKED((mask & input(row, col, 0, digit, 0).Raw()) << ch_low);
+            }
+          }
+  } else {
+    for(T_UINT ih = 0; ih < input_depth; ++ih)
+      for(T_UINT wi = 0; wi < out_height; wi++)
+        for(T_UINT wj = 0; wj < out_width; wj++)
+          for(T_UINT ki = 0; ki < kernel_size; ki++)
+            for(T_UINT kj = 0; kj < kernel_size; kj++)
+            {
+              T_INT row = (wi * stride) + ki;
+              T_INT col = (wj * stride) + kj;
+              for(T_UINT digit = 0; digit < bits_per_input; ++digit) {
+                output(wi, wj,
+                  ih + (ki * kernel_size + kj) * input_depth,
+                  digit, 0) = input(row, col, ih, digit, 0);
```

Comment:
Could we introduce additional variable here, and reformat it a bit, to make it a bit easier to comprehend?

E.g
```cpp
                T_UINT ch_high = ih + (ki * kernel_size + kj) * input_depth;
                output(wi, wj, ch_high, digit, 0) = input(row, col, ih, digit, 0);
```

Reply:
Fixed.

Reply:
Thank you!